code/script.js

```diff
@@ -64,33 +64,39 @@ do {
     4 - Caesar Salad`);
       subtypeChoiceFood = ["Garden Salad", "Greek Salad", "Caprese Salad", "Caesar Salad"];
       break;
-    default: //i don't think I need this? If an invalid option is chosen, show an error message and exit the program. //alert("Invalid choice."); //process.exit(1);
+    default:
+      alert('Your choice is not available, Please try again');
   }
 }
-while (subtypeChoiceNum >= 5); //do...while loop if user does not enter a valid option (1,2 or 3)
+while (subtypeChoiceNum < 1 || subtypeChoiceNum > 4); //do...while loop if user does not enter a valid option (1,2, 3 or 4) they are asked again. It does not work if user inputs text. Diego has helped with tips and advice on how to check if a variable is a number.
 
 alert(`Great Choice, You chose Number ${subtypeChoiceNum} - ${subtypeChoiceFood[subtypeChoiceNum - 1]}`);
 
 // Step 4 - Age
 let userAgeAlert = "Adult"
 const userAgeNum = prompt("So that we can prepare the correct size. Please tell us your age?")
+if (isNaN(userAgeNum))
+  alert('Invalid input. Please enter a valid number (1, 2, or 3)');
 if (userAgeNum >= 18) {
   alert("You are an adult. We will prepare you an adult size portion")
   userAgeAlert = "Adult"
-} else {
+} else if (userAgeNum >= 1 || userAgeNum < 18) {
   alert("You are a child. We will prepare you an child size portion")
   userAgeAlert = "Child"
+} else {
+  alert("Your input is invaild, please try again")
```

Comment:
You might not need both the "alert('Invalid input. Please enter a valid number (1, 2, or 3)');" and the "alert("Your input is invaild, please try again")". When I enter a letter instead of a number I get both alerts, one after the other.

Edit: I see that you only have one alert in the "switch.js", but both show up when I test.